packages/generate/src/queries.ts

```diff
@@ -147,9 +144,12 @@ currently supported.`);
   //   generate output file
 }
 
-export function stringifyImports(imports: { [k: string]: boolean }) {
-  if (Object.keys(imports).length === 0) return "";
-  return `import type {${Object.keys(imports).join(", ")}} from "edgedb";`;
+export function stringifyImports(imports: $.ImportMap) {
+  return [...imports]
+    .map(([module, specifiers]) => {
+      return `import type {${[...specifiers].join(", ")}} from "${module}";`;
+    })
```

Comment:
Oh my, I love this. One tiny non-blocking nit:

```suggestion
    .map(([module, specifiers]) =>
      `import type {${[...specifiers].join(", ")}} from "${module}";`
    )
```

This line is kinda noisy already, so this feels a little easier on my eyes/brain now that we don't have any branching logic.

Reply:
This is my own nit as well. I think it was late.